cnxarchive/database.py

```diff
@@ -186,9 +196,14 @@ def get_collections(module_ident, plpy):
         FROM trees c JOIN t ON (c.nodeid = t.parent)
         WHERE not c.nodeid = ANY(t.path)
     )
-    SELECT DISTINCT m.module_ident
-    FROM t JOIN latest_modules m ON (t.document = m.module_ident)
-    WHERE t.parent IS NULL ORDER BY m.module_ident
+    SELECT * FROM (
+        SELECT DISTINCT m.module_ident, m.revised,
+            MAX(m.revised) OVER (PARTITION BY m.uuid) as latest
+        FROM t, modules m
+        WHERE t.parent IS NULL and t.document = m.module_ident
+    ) r
+    WHERE r.revised = r.latest
+    ORDER BY r.module_ident
```

Comment:
I tried to do something like:

```
with t as (
    select 188037 as documentid union all select 97821 as documentid
), uuids as (
    select uuid from modules m join t on t.documentid = m.module_ident
), max_versions as (
    select max(module_ident) from modules m join uuids on m.uuid = uuids.uuid group by m.uuid
) select * from max_versions;
```

It is kind of working, but let's see how yours look :D